api/src/types/s2-api.ts

```diff
@@ -17,3 +17,11 @@ interface S2ApiAuthor {
   name: string;
   url: string;
 }
+
+export interface S2ApiError {
+  error: string;
+}
+
+export function isS2ApiError(d: any): d is S2ApiError {
```

Comment:
Oh wow! Is that a way to assert something about the code? I would have assumed this returned `boolean`, so that's pretty awesome!

Reply:
Type guards! https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

It's effectively returning a `boolean`, yeah, but to TypeScript, the boolean tells the type-checker that it can assume exactly the type of the object within that scope. It's usually used like:
```typescript
if (isFoo(foo)) {
  foo.onlyFooThing(); // this is good!
}
(foo as Foo).onlyFooThing(); // casting required here
```